modules/poetry.nix

```diff
@@ -19,9 +21,78 @@
   }: let
     args = {inherit self system lib config name kernelName requiredRuntimePackages;};
     kernelModule = import ./kernel.nix args;
+
+    kernelFunc = {
+      self,
+      system,
+      # custom arguments
+      pkgs,
+      name,
+      argvKernelName,
+      codemirrorMode,
+      displayName,
+      language,
+      requiredRuntimePackages,
+      runtimePackages,
+      # https://github.com/nix-community/poetry2nix
+      poetry2nix,
+      poetry,
+      # https://github.com/nix-community/poetry2nix#mkPoetryPackages
+      projectDir,
+      pyproject,
+      poetrylock,
+      overrides,
+      python,
+      editablePackageSources,
+      extraPackages,
+      preferWheels,
+      groups,
+      ignoreCollisions,
+    }: let
+      env =
```

Comment:
we should move the env to the moduleType totally, and make sure the users are able to rescue the poetyEnv and pure python support. like that https://github.com/GTrunSec/jupyterWith/blob/dev/modules/types/poetry.nix#L136

Reply:
fixes https://github.com/tweag/jupyenv/issues/485